File: cpp/gabaritos/gabarito18.cpp
#include <iostream>
#include <vector>

int soma_recursiva(const std::vector<int>& lst, size_t index = 0) {
    if (index == lst.size()) return 0;
    return lst[index] + soma_recursiva(lst, index + 1);
}

int main() {
    std::vector<int> lista = {1, 2, 3, 4, 5};
    std::cout << "Soma: " << soma_recursiva(lista) << std::endl;
    return 0;
}

File: csharp/gabaritos/gabarito18.cs
using System;
using System.Collections.Generic;

class Program
{
    static int SomaRecursiva(List<int> lst, int index)
    {
        if (index == lst.Count)
            return 0;
        return lst[index] + SomaRecursiva(lst, index + 1);
    }

    static void Main()
    {
        List<int> lista = new List<int> { 1, 2, 3, 4, 5 };
        Console.WriteLine($"Soma: {SomaRecursiva(lista, 0)}");
    }
}